### DISCUSSION TITLE
support for riscv compliance test suite for gem5 simulator

### DISCUSSION THREAD

**Discussion by rosekunthara** [gem5/gem5#1392] [Q&A]
Hello,
Is it possible to run riscv compliance test suite for gem5 simulator?
When I ran executable file, like dhrystone.riscv, in gem5 SE mode, core dump error occurs like below.

Global frequency set at 1000000000000 ticks per second
warn: No dot file generated. Please install pydot to generate the dot file and pdf.
src/mem/dram_interface.cc:690: warn: DRAM device capacity (8192 Mbytes) does not match the address range assigned (1024 Mbytes)
src/arch/riscv/isa.cc:276: info: RVV enabled, VLEN = 256 bits, ELEN = 32 bits
src/arch/riscv/linux/se_workload.cc:60: warn: Unknown operating system; assuming Linux.
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
system.remote_gdb: Listening for connections on port 7002
Beginning simulation!
src/sim/simulate.cc:199: info: Entering event queue @ 0.  Starting simulation...
src/arch/riscv/faults.cc:213: panic: Illegal instruction 0x1003002a073 at pc (0x80000044=>0x80000048).(0=>1): mstatus is not accessible in 0
Memory Usage: 1158348 KBytes
Program aborted at tick 1722000
--- BEGIN LIBC BACKTRACE ---
./build/RISCV/gem5.opt(_ZN4gem515print_backtraceEv+0x30)[0x563813af05a0]
./build/RISCV/gem5.opt(_ZN4gem512abortHandlerEi+0x4c)[0x563813b1596c]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x14420)[0x7f8f30429420]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x7f8f2fce800b]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x7f8f2fcc7859]
./build/RISCV/gem5.opt(+0xd63585)[0x563813023585]
./build/RISCV/gem5.opt(_ZN4gem58RiscvISA16IllegalInstFault8invokeSEEPNS_13ThreadContextERKNS_14RefCountingPtrINS_10StaticInstEEE+0xcc)[0x56381347fd7c]
./build/RISCV/gem5.opt(_ZN4gem58RiscvISA10RiscvFault6invokeEPNS_13ThreadContextERKNS_14RefCountingPtrINS_10StaticInstEEE+0x2e7)[0x563813480247]
./build/RISCV/gem5.opt(_ZN4gem513BaseSimpleCPU9advancePCERKSt10shared_ptrINS_9FaultBaseEE+0xde)[0x5638131238ee]
./build/RISCV/gem5.opt(_ZN4gem515TimingSimpleCPU11advanceInstERKSt10shared_ptrINS_9FaultBaseEE+0xc6)[0x563813115776]
./build/RISCV/gem5.opt(_ZN4gem515TimingSimpleCPU14completeIfetchEPNS_6PacketE+0x2b8)[0x563813116798]
./build/RISCV/gem5.opt(_ZN4gem510EventQueue10serviceOneEv+0x175)[0x563813b06485]
./build/RISCV/gem5.opt(_ZN4gem59doSimLoopEPNS_10EventQueueE+0x70)[0x563813b2f910]
./build/RISCV/gem5.opt(_ZN4gem58simulateEm+0x28b)[0x563813b2ff9b]
./build/RISCV/gem5.opt(+0x1f36a90)[0x5638141f6a90]
./build/RISCV/gem5.opt(+0xd76498)[0x563813036498]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(+0x2a8748)[0x7f8f306e0748]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(_PyEval_EvalFrameDefault+0x8dd8)[0x7f8f304b5f48]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(_PyEval_EvalCodeWithName+0x8fb)[0x7f8f30602e4b]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(_PyFunction_Vectorcall+0x94)[0x7f8f306e0124]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(+0x74d6d)[0x7f8f304acd6d]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(_PyEval_EvalFrameDefault+0x7d86)[0x7f8f304b4ef6]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(_PyEval_EvalCodeWithName+0x8fb)[0x7f8f30602e4b]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(PyEval_EvalCodeEx+0x42)[0x7f8f306031d2]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(PyEval_EvalCode+0x1f)[0x7f8f306035bf]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(+0x1cfc01)[0x7f8f30607c01]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(+0x25f547)[0x7f8f30697547]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(+0x74d6d)[0x7f8f304acd6d]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(_PyEval_EvalFrameDefault+0x12fd)[0x7f8f304ae46d]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(+0x8006b)[0x7f8f304b806b]
/lib/x86_64-linux-gnu/libpython3.8.so.1.0(PyVectorcall_Call+0x60)[0x7f8f306e0840]
./build/RISCV/gem5.opt(+0x1856a24)[0x563813b16a24]
--- END LIBC BACKTRACE ---
For more info on how to address this issue, please visit https://www.gem5.org/documentation/general_docs/common-errors/ 

Aborted (core dumped)

**Comment by rogerchang23424** [comment]
These files need to run in FS mode, please use `config/example/riscv/fs_linux.py`
